File: 2020/aoc2020/src/test/kotlin/Day21Test.kt
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

@DisplayName("Day 21")
class Day21Test {
    private val example = readText("day21_example.txt")
    private val input = readText("day21.txt")
    private val day = Day21()

    @Nested
    @DisplayName("Part 1")
    inner class Part1 {
        @Test
        fun `Matches example`() {
            val answer = day.part1(example)
            assertEquals(5, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part1(input)
            assertEquals(2595, answer)
        }
    }

    @Nested
    @DisplayName("Part 2")
    inner class Part2 {
        @Test
        fun `Matches example`() {
            val answer = day.part2(example)
            assertEquals("mxmxvkd,sqjhc,fvjkl", answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part2(input)
            assertEquals("thvm,jmdg,qrsczjv,hlmvqh,zmb,mrfxh,ckqq,zrgzf", answer)
        }
    }
}

File: 2024/AdventOfCode/Tests/Day21Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day21Test {
    private const string Example = "Inputs/21-Example.txt";
    private const string Input = "Inputs/21.txt";
    private static Day21 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day21(Example);
            const string expected = "126384";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day21(Input);
            const string expected = "248108";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day21(Example);
            const string expected = "154115708116294";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day21(Input);
            const string expected = "303836969158972";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}